### DISCUSSION TITLE
Difference between 'agent-token' in config.yaml and token in node-token file

### DISCUSSION THREAD

**Discussion by zulh-civo** [k3s-io/k3s#6873] [Q&A]
What's the difference between the `agent-token` in `/etc/rancher/k3s/config.yaml` file and the token returned by `/var/lib/rancher/k3s/server/node-token` file? What are their use-cases?

**Comment by brandond** [comment]
https://github.com/k3s-io/k3s/issues/6175#issuecomment-1257418209

**Reply by zulh-civo** [reply]
Thanks. What are the pros and cons of bare token vs K10 format? How can we tell if a server/cluster is using bare token or K10? From my observation, both exist in the server/master node.

**Reply by brandond** [reply]
> What are the pros and cons of bare token vs K10 format?

The full format pins a root of trust for the server being joined, ensuring that the credentials aren't being sent to any host other than one in the cluster you got the token from. The code flow for this is discussed in the issue I linked.

> How can we tell if a server/cluster is using bare token or K10

Look at the config file or CLI flags, see which one you've used.
